.github/workflows/pr-release-workflow.yml

```diff
@@ -0,0 +1,29 @@
+name: PR Release Workflow
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    types:
+      - milestoned
+      - closed
+
+jobs:
+  backport-commit:
+    runs-on: ubuntu-22.04
+    if: >-
+      github.repository == 'llvm/llvm-project' &&
+      github.event.pull_request.merged &&
+      contains(github.event.pull_request.milestone.title, 'Release')
```

Comment:
This doesn't check that the PR targets the current Release milestone. So, AFAICT, if I create a milestone for the next release (LLVM 21) and I merge a PR, it would get automatically cherry-picked onto the release branch. I don't think that's intended.

Reply:
They cherry-pick step would fail, because the github-automation.py script looks for the branch name in the description of the milestone.  So if the milestone was for LLVM 21, it would try to cherry-pick to release/21.x and fail.  If this happened, there wouldn't be a very descriptive error message, so we could do something to make the error reason more clear.